impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -391,6 +403,69 @@ public ClientBuilder setPrivateKey(String privateKey) {
         return this;
     }
 
+    @Override
+    public ClientBuilder setPrivateKey(Path privateKeyPath) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(privateKeyPath, "Missing privateKeyFile");
+            this.clientConfig.setPrivateKey(getFileContent(privateKeyPath));
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setPrivateKey(InputStream privateKeyStream) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(privateKeyStream, "Missing privateKeyFile");
+            this.clientConfig.setPrivateKey(getFileContent(privateKeyStream));
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setPrivateKey(PrivateKey privateKey) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(privateKey, "Missing privateKeyFile");
+            String encodedString = "-----BEGIN PRIVATE KEY-----\n";
+            encodedString = encodedString + Base64.getEncoder().encodeToString(privateKey.getEncoded()) + "\n";
+            encodedString = encodedString + "-----END PRIVATE KEY-----\n";
+            this.clientConfig.setPrivateKey(encodedString);
+        }
+        return this;
+    }
+
+    private String getFileContent(File file) {
+        try (InputStream inputStream = new FileInputStream(file)) {
+            return readFromInputStream(inputStream);
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage());
```

Comment:
throw new IllegalArgumentException("Could not read from supplied private key file")

Reply:
fixed